src/screens/Wallet.js

```diff
@@ -109,10 +166,81 @@ class Wallet extends React.Component {
       );
     }
 
+    const renderTabAdmin = () => {
+      if (this.shouldShowAdministrativeTab()) {
+        return (
+            <li className="nav-item">
+              <a className="nav-link" id="administrative-tab" data-toggle="tab" href="#administrative" role="tab" aria-controls="administrative" aria-selected="false">Administrative Tools</a>
+          </li>
+        );
+      } else {
+        return null;
+      }
+    }
+
+    const renderContentAdmin = () => {
+      if (this.shouldShowAdministrativeTab()) {
+        return (
+          <div className="tab-pane fade" id="administrative" role="tabpanel" aria-labelledby="administrative-tab">
+            <TokenAdministrative token={token} />
+          </div>
+        );
+      } else {
+        return null;
+      }
+    }
+
+    const renderTokenData = (token) => {
+      if (this.props.selectedToken === hathorLib.constants.HATHOR_TOKEN_CONFIG.uid) {
+        return renderWallet();
+      } else {
+        return (
+          <div>
+            <ul className="nav nav-tabs mb-4" id="tokenTab" role="tablist">
+              <li className="nav-item">
+                <a className="nav-link active" id="wallet-tab" data-toggle="tab" href="#wallet" role="tab" aria-controls="home" aria-selected="true">Balance & History</a>
+              </li>
+              <li className="nav-item">
+                <a className="nav-link" id="token-tab" data-toggle="tab" href="#token" role="tab" aria-controls="token" aria-selected="false">About {token.name}</a>
+              </li>
+              {renderTabAdmin()}
+            </ul>
+            <div className="tab-content" id="tokenTabContent">
+              <div className="tab-pane fade show active" id="wallet" role="tabpanel" aria-labelledby="wallet-tab">
+                {renderWallet()}
+              </div>
+              <div className="tab-pane fade" id="token" role="tabpanel" aria-labelledby="token-tab">
+                <TokenGeneralInfo token={token} showConfigString={true} errorMessage={this.state.errorMessage} />
+              </div>
+              {renderContentAdmin()}
+            </div>
+          </div>
+        );
+      }
+    }
+
+    const getUnregisterBody = () => {
+      if (token === undefined) return null;
+
+      return (
+        <div>
+          <p>Are you sure you want to unregister the token <strong>{token.name} ({token.symbol})</strong>?</p>
+          <p>You won't lose your tokens, you just won't see this token on the side bar anymore.</p>
+        </div>
+      )
+    }
+
     const renderUnlockedWallet = () => {
+      const isHathorToken = this.props.selectedToken === hathorLib.constants.HATHOR_TOKEN_CONFIG.uid;
```

Comment:
It appears in so many places that we should consider creating an util method for it.